engine/class_modules/sc_death_knight.cpp

```diff
@@ -10405,6 +10406,24 @@ void death_knight_t::init_finished()
   }
 }
 
+// death_knight_t::validate_fight_style =====================================
+
+bool death_knight_t::validate_fight_style( fight_style_e style ) const
+{
+  if ( specialization() == DEATH_KNIGHT_BLOOD )
+  {
+    switch ( style )
+    {
+    case FIGHT_STYLE_DUNGEON_ROUTE:
+    case FIGHT_STYLE_DUNGEON_SLICE:
+      return false;
+    default:
+      return true;
+    }
+  }
+  return true;
+}
```

Comment:
```suggestion
bool death_knight_t::validate_fight_style( fight_style_e fight ) const
{
  if ( specialization() == DEATH_KNIGHT_BLOOD )
  {
    switch ( fight )
    {
    case FIGHT_STYLE_PATCHWERK:
    case FIGHT_STYLE_CASTING_PATCHWERK:
      return true;
    default:
      return false;
    }    
  }
  if ( specialization() == DEATH_KNIGHT_FROST )
  {
    switch ( fight )
    {
    case FIGHT_STYLE_PATCHWERK:
    case FIGHT_STYLE_CASTING_PATCHWERK:
    case FIGHT_STYLE_DUNGEON_SLICE:
    case FIGHT_STYLE_DUNGEON_ROUTE:
      return true;
    default:
      return false;
    }
  }

  if ( specialization() == DEATH_KNIGHT_UNHOLY )
  {
    switch ( fight )
    {
    case FIGHT_STYLE_PATCHWERK:
    case FIGHT_STYLE_CASTING_PATCHWERK:
    case FIGHT_STYLE_DUNGEON_SLICE:
    case FIGHT_STYLE_DUNGEON_ROUTE:
      return true;
    default:
      return false;
    }
  }
  return false;
}
```

I'd like to leverage `validate_fight_style` for DPS specs as well and I think the easiest way would be to drop the new function into your PR.

Reply:
Manually corrected for in conjunction with combining changes from https://github.com/simulationcraft/simc/pull/8665.